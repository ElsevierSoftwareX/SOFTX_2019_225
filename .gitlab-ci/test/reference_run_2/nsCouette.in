&parameters_grid
m_r   = 32                 ! radial points           => m_r      grid points (radial)
m_th  = 384                ! azimuthal Fourier modes => 2*m_th+1 grid points (azimuthal)
m_z0  = 320                ! axial Fourier modes     => 2*m_z0+1 grid points (axial)
k_th0 = 1.0d0              ! azimuthal wavenumber    => L_th = 2*pi/k_th0 azimuthal length of grid
k_z0  = 3.14159265358979d0 ! axial wavenumber        => L_z =  2*pi/k_z0  axial length of grid
eta   = 0.5d0              ! aspect ratio => r_i = eta/(1-eta),r_o = 1/(1-eta) inner/outer radius
/

&parameters_physics
Re_i = 111.1d0         ! Inner cylinder Reynolds number
Re_o =   0.0d0         ! Outer cylinder Reynolds number
Gr   = 280.0d0         ! Grashof number
Pr   =   15.d0         ! Prandtle number
gap  =    3.25d0       ! gap size in cm                          [TE_CODE only]
gra  =  980.0d0        ! gravitational acceleration in g/cm**3   [TE_CODE only]
nu   =    1.01d-2      ! kinematic viscosity in cm**2 /s         [TE_CODE only]
/

&parameters_timestep
numsteps    = 100       ! number of steps 
init_dt     = 1.0d-4    ! initial size of timestep
variable_dt = T         ! use a variable (=T) or fixed (=F) timestep
maxdt       = 1.0d-2    ! maximum size of timestep
Courant     = 0.25d0    ! CFL safety factor
/

&parameters_output
fBase_ic = 'REF_2'      ! identifier for coeff_ (checkpoint) and fields_ (hdf5) files 
dn_coeff = 5000         ! output interval [steps] for coeff (dn_coeff = -1 disables ouput)
dn_ke    = 10           ! output interval [steps] for energy
dn_vel   = 10           ! output interval [steps] for velocity
dn_Nu    = 10           ! output interval [steps] for Nusselt (torque)
dn_hdf5  = 1000         ! output interval [steps] for HDF5 output
print_time_screen = 10  ! output interval [steps] for timestep info to stdout 
/

&parameters_control
restart = 0		! initialization mode: 0=new run, 1=restart from checkpoint
runtime = 7200          ! maximum runtime [s] for the job
/

&parameters_initialcondition
ic_tcbf    = T                ! Set Taylor-Couette base flow (T) or resting fluid (F), only when restart = 0
ic_temp    = T                ! Set temperature profile (T) or zero (F), only when restart = 0, only TE_CODE
ic_pert    = T                ! Add perturbation on top of base flow (T) or not (F), only when restart = 0
ic_p(1, :) = 1.0d-2, 0, 2     ! 1st perturbation: amplitude and wavevector (a1, k_th1, k_z1)
ic_p(2, :) = 1.0d-2, 3, 0     ! 2nd perturbation: amplitude and wavevector (a2, k_th2, k_z2)
ic_p(2, :) = 1.0d-2, 3, 0     ! 2nd perturbation: amplitude and wavevector (a2, k_th2, k_z2)
ic_p(3, :) = 0.0d-2, 0, 0     ! 3rd perturbation: amplitude and wavevector (a3, k_th3, k_z3)
ic_p(4, :) = 0.0d-2, 0, 0     ! 4th perturbation: amplitude and wavevector (a4, k_th4, k_z4)
ic_p(5, :) = 0.0d-2, 0, 0     ! 5th perturbation: amplitude and wavevector (a5, k_th5, k_z5)
ic_p(6, :) = 0.0d-2, 0, 0     ! 6th perturbation: amplitude and wavevector (a6, k_th6, k_z6)
/
