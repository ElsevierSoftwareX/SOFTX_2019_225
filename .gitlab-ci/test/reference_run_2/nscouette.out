!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
! NSCouette, a HPC code for DNS of Taylor-Couette flow                 !
!                                                                      !
! Copyright (C) 2016 Marc Avila, Bjoern Hof, Jose Manuel Lopez,        !
!                    Markus Rampp, Liang Shi                           !
!                                                                      !
! NSCouette is free software: you can redistribute it and/or modify    !
! it under the terms of the GNU General Public License as published by !
! the Free Software Foundation, either version 3 of the License, or    !
! (at your option) any later version.                                  !
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

starting NSCouette (ed647dbe) using    4 MPI tasks,    2 OpenMP threads/task

   TASKS:   4 THREADS: 2    THIS:   0     Host:buildtest- Cores:   0   1
   TASKS:   4 THREADS: 2    THIS:   1     Host:buildtest- Cores:   2   3
   TASKS:   4 THREADS: 2    THIS:   2     Host:buildtest- Cores:   4   5
   TASKS:   4 THREADS: 2    THIS:   3     Host:buildtest- Cores:   6   7
  step=          10  dt=  5.449244683038102E-006
  step=          20  dt=  5.449244683038102E-006
  step=          30  dt=  5.449244683038102E-006
  step=          40  dt=  5.449244683038102E-006
  step=          50  dt=  5.449244683038102E-006
  step=          60  dt=  5.449244683038102E-006
  step=          70  dt=  5.449244683038102E-006
  step=          80  dt=  5.449244683038102E-006
  step=          90  dt=  5.449244683038102E-006
  step=         100  dt=  5.449244683038102E-006
 written coeff file to disk: coeff_REF_2.00000100                    
T0: number of timesteps              :   100
T0: total time incl I/O [s]:   1742.20
T0:  av time per timestep excl I/O [s]:     17.02
ALL: min/max av time per timestep [s]:   17.0165   17.0184
 /= Group                                              [s]     fraction
 |                                            ============ ============
 |_ [Root] (running)                           1742.333978        1.000
   |_ (own)                                         0.000233        0.000
   |_ main                                       1742.333734        1.000
     |_ (own)                                         1.406163        0.001
     |_ pretstp                                       3.880664        0.002
     |_ tstep                                      1704.470755        0.978
       |_ (own)                                        30.446985        0.018
       |_ nonlin  (200x)                             1446.541715        0.849
         |_ (own)                                         0.305201        0.000
         |_ matvec  (200x)                               33.395142        0.023
         |_ tranf  (200x)                                54.993226        0.038
         |_ deriv  (200x)                               709.333530        0.490
         |_ fft  (200x)                                 507.673738        0.351
         |_ tranb  (200x)                               140.840878        0.097
       |_ pretstp  (2x)                                 4.043425        0.002
       |_ predic  (100x)                              105.556446        0.062
       |_ correc  (100x)                              115.060128        0.068
         |_ (own)                                       103.228326        0.897
         |_ mcorr  (100x)                                11.831802        0.103
       |_ io energy  (10x)                              2.822056        0.002
     |_ io coeff                                     32.576152        0.019
